src/harness/sas_test_harness.py

```diff
@@ -0,0 +1,364 @@
+    full_activity_dump['generationDateTime'] = end_time
+    full_activity_dump['description'] = 'Load {} FAD to SAS UUT'.format(self.host_name)
+    return full_activity_dump
+
+  def __createFadRecord(self, encoded_url, record):
+    """ Creates the files object that includes url,checksum,size,version and recordtype of record
+        required to be packed in FAD.
+    """
+    fad_record = {}
+    fad_record['url'] = encoded_url
+    fad_record['checksum'] = hashlib.sha1(json.dumps(record)).hexdigest()
+    fad_record['size'] = sys.getsizeof(record)
+    fad_record['version'] = self.sas_version
+    if 'cbsd' in encoded_url:
+      fad_record['recordType'] = "cbsd"
+    elif 'zone' in encoded_url:
+      fad_record['recordType'] = "zone"
+    elif 'esc_sensor' in encoded_url:
+      fad_record['recordType'] = "esc_sensor"
+    else:
+      raise Exception("ConfigurationError:: Incorrect Record Type received in URL:{}".format(encoded_url))
+    return fad_record
+
+  def __writeDumpFile(self, file_name, data):
+    """ Write JSON data to specified file. """
+    dump_file_path = os.path.join(self.getDumpFilePath(), file_name)
+    with open(dump_file_path, "w") as file_handler:
+      file_handler.writelines(json.dumps(data, sort_keys=True, indent=4))
+
+  def writeFadRecords(self, dump_records_list):
+    """ This method is used to write dump records in specified path these files will be served
+        by SAS Test Harness.
+    Args:
+       dump_records_list: List of dump_records object required to be written on the dump path.
+
+    Returns:
+       It returns None and all records are dumped in specified path.
+    """
+    fad_record_list = []
+    generated_file_name = None
+    record_url = None
+    dump_path = self.getDumpFilePath()
+
+    for dump_records in dump_records_list:
+      for index, dump_record in enumerate(dump_records):
+        if dump_record.has_key('id'):
+          if 'cbsd' in dump_record['id']:
+            generated_file_name = '{}.json'.format(dump_record['id'].decode('utf-8').split('/')[-1])
+            record_url = self.getBaseUrl() + '/cbsd/%s' % quote(dump_record['id'])
+          elif 'zone' in dump_record['id']:
+            generated_file_name = '{}.json'.format(dump_record['id']).split('/')[-1]
+            record_url = self.getBaseUrl() + '/' + dump_record['id']
+          elif 'esc_sensor' in dump_record['id']:
+            encoded_sensor_id = str(hashlib.sha1(dump_record['id']).hexdigest()).encode('utf-8')
+            generated_file_name = '{}.json'.format(encoded_sensor_id)
+            record_url = self.getBaseUrl() + '/%s/%s' % (dump_record['id'], encoded_sensor_id)
+          else:
+            raise Exception("ConfigurationError:incorrect record type. Unable to write FAD record")
+        else:
+          raise Exception("ConfigurationError:field id is not found. Unable to write FAD record")
+        if record_url is not None and generated_file_name is not None:
```

Comment:
Style note: you can use this structure to make it more readable:

```
if not record_url or not generated_file_name:
  raise Exception()

self.__writeDumpFile(generated_file_name, dump_record)
...
```

It reads a lot more like English, it reduces the indentation level for the main logic, and it reduces the number of branches that your brain has to keep track of -- all of which help readability tremendously.

I think this code will end up changing anyway, but it's something worth keeping in mind for the future.

Reply:
Updated   correctly